### ISSUE TITLE
[Bug] False positive for [E0277] 

### ISSUE THREAD

**Issue by Lim3zer0** [intellij-rust/intellij-rust#6331]
## Environment

Plugin Version: 0.3.133.3426-202
IDE:  CLion Commercial 2020.2.4 
Rust Toolchain: 1.47.0 stable-x86_64-unknown-linux-gnu
OS: Fedora 33 KDE

## Problem description / Steps to reproduce
False positive for [E0277] 
The code below, which can be compiled successfully, demonstrates the issue:
```rust
trait Foo {}

trait Bar: Foo {}

trait Baz: Foo {}

trait Bif: Baz + Bar {}

impl<T> Baz for T where T: Foo {}

impl<T> Bif for T where T: Bar {}
```
In this example `Bif` will be incorrectly highlighted by CLion to say [E0277] because the trait bound Baz is not satisfied.
This is incorrect. 
The reason being is as follows: 
- In `impl Bif` `T` is already bound by `Bar`, and any implementation of `Bar` must implement `Foo`
- `Baz` is is automatically implemented for any `T` where `T` implements `Foo`

Because of these two things it is guaranteed that when a type `T` is bound by the trait `Bar` it must have an implementation of `Baz`and thus the trait bound of `Baz` is satisfied.

The problem can be 'solved' by changing the last line to this:
```
impl<T> Bif for T where T: Bar + Baz {}
```
however this is not needed for rustc to compile the code.

**Comment by Kobzol** [comment]
@vlad20012 it seems that here:
```rust
trait Foo {}
trait Baz: Foo {}

impl<T> Foo for T {}
impl<T> Baz for T {}
```
`ImplLookup` cannot see that it is valid, it thinks that `T` does not implement `Foo` in the second impl. Is this intended for the current lookup implementation or am I using lookup wrong? https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt#L675

**Comment by hi2u** [comment]
Here's another example of E0277 false positives (code compiles fine without any warnings/errors):

![2020-11-19_024957-Window](https://user-images.githubusercontent.com/9689599/99561694-f863cc80-2a12-11eb-8039-032770e1d6e4.png)

...not sure if the HTML entities thing is tripping it up in this case, as it doesn't complain about any of the other trait dependencies:

    Unpin
    + serde::export::fmt::Debug
    + std::marker::Send

Haven't also posted the screenshot on https://github.com/intellij-rust/intellij-rust/issues/5241 about the HTML entities display issue.

* Rust plugin version: 0.3.135.3484-203

* IntelliJ IDEA 2020.3 Beta (Ultimate Edition)
```
- Build #IU-203.5784.10, built on November 10, 2020
- IntelliJ IDEA EAP User
- Runtime version: 11.0.9+11-b1145.10 amd64
- VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
- Windows 10 10.0
- GC: ParNew, ConcurrentMarkSweep
- Memory: 1981M
- Cores: 8
- Non-Bundled Plugins: Statistic, com.4lex4.intellij.solarized, intellij.haskell, com.markskelton.one-dark-theme, in.xiv.cs, org.toml.lang, com.jetbrains.php, training, org.rust.lang, com.samdark.intellij-visual-studio-code-dark-plus
```

**Comment by Undin** [comment]
@hi2u looks like your case is not related to original issue. Most likely it's because of proc-macro `derive(sqlx::FromRow)`. Unfortunately, proc macros are not supported by the plugin for now. See #1786
